traits/ctraits.c

```diff
@@ -580,6 +580,39 @@ set_value ( PyObject ** field, PyObject * value ) {
     return 0;
 }
 
+static PyObject *
```

Comment:
It would be good to have a docstring-style comment above this function describing what it does. (And the same for the other new functions.)